ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/streaming/StreamedQueryResource.java

```diff
@@ -489,6 +491,18 @@ private EndpointResponse handleTablePullQuery(
       pullBandRateLimiter.allow(KsqlQueryType.PULL);
 
       final Optional<Decrementer> optionalDecrementer = Optional.ofNullable(decrementer);
+      Optional<ConsistencyOffsetVector> consistencyOffsetVector = Optional.empty();
+      if (ksqlConfig.getBoolean(KsqlConfig.KSQL_QUERY_PULL_CONSISTENCY_OFFSET_VECTOR_ENABLED)
```

Comment:
Just to make sure I understand, if the client doesn't send up anything, the feature is not enabled, right?

There are four main cases:
- The feature is disabled on the server
- The feature is enabled on the server, but not the client
- The feature is enabled on the server and the client, but the client has provided no consistency token
- The feature is enabled on the server and client, and the client provides a consistency token for the pull query.

This seems to cover all cases, but I just want to be sure.  2 is covered because the client indicates that the feature is enabled but no consistency token is provided by using an empty string?

Reply:
The client needs to set the config to true and then only will the consistency token be included in the request properties. So, if it is disabled on the client, no consistency token will be added to the request properties and the server won't do anything.
The empty string is used only when it is enabled and signals an empty, not yet initialized consistency token.